src/features/Admin/Facilities/Facilities.tsx

```diff
@@ -23,51 +27,59 @@ const Facilities = () => {
     { label: "Created At", key: "createdAt" },
     { label: "Updated At", key: "updatedAt" },
   ];
+  const [facilityId, setFacilityId] = useState(0);
+  const [facilityDetails, setFacilityDetails] = useState([]);
+  // Modal
+  const [modalState, setModalState] = useState("close");
+  const handleClose = () => setModalState("close");
 
-  // const handleOpenDialog = () => {
-  //   setDialogOpen(true);
-  // };
+  // Add_Modal
+  const showAddModal = () => {
+    // setFacilityId(facilityId);
+    setModalState("add-modal");
+  };
+  // view-Modal
+  const showViewModal = (id) => {
+    // console.log(id);
+    setFacilityId(id);
+    setModalState("view-modal");
+    getFacilityDetails(id);
+  };
+  // Update-Modal
+  const showUpdateModal = (facility) => {
+    setFacilityId(facility._id);
+    setValue("name", facility.name);
 
-  // const handleCloseDialog = () => {
-  //   setDialogOpen(false);
-  // };
+    setModalState("update-modal");
+  };
+  // Delete_Modal
+  const showDeleteModal = (facilityId) => {
+    setFacilityId(facilityId);
+    setModalState("delete-modal");
+  };
 
-  // const showViewModal = (itemId) => {
-  //   handleOpenDialog(itemId);
-  //   handleOpenDialog();
-  // };
   interface IFacility {
     name: string;
   }
   const {
     register,
     handleSubmit,
+    setValue,
     formState: { errors },
   } = useForm<IFacility>();
 
-  //********** Generic function to open dialog based on action
-  const handleOpenDialog = (action, item = null) => {
-    setDialogAction(action);
-    setSelectedItem(item);
-    setDialogOpen(true);
-  };
-  const handleCloseDialog = () => {
-    setDialogOpen(false);
-  };
-  
-
   // ************Add Facility
   const onSubmit: SubmitHandler<IFacility> = async (data: IFacility) => {
     // console.log(data);
     console.log("Valeur de name avant envoi:", data.name);
 
     await axios
-      .post(`${facilitiesUrl}`, data, {
+      .post(`${facilitiesRoomsUrl}`, data, {
         headers: requestHeaders,
       })
       .then((response) => {
         console.log("succ response", response);
-        handleCloseDialog();
+        handleClose();
         getAllFacilities();
```

Comment:
consider using the `useFacilities` hook to get the facilities related data such as facilities, loading state, error message, pagination. This will reduce the logic written in this file.